doc/contributing.rst

```diff
@@ -1 +1,191 @@
+contributions in the form of bug reports, documentation, code, feature requests,
+and more. These guidelines provide resources on how best to contribute.
+
+For new users, checking out the `GitHub guides <https://guides.github.com>`_ are
+recommended.
+
+.. Many of these steps follow napari's contributor guide:
+   https://github.com/napari/napari/blob/master/docs/CONTRIBUTING.md
+
+.. _setting-up-a-development-installation:
+
+Setting up a development installation
+=====================================
+
+You need a `fork <https://guides.github.com/activities/forking/#fork>`_ of the
+`repository <https://github.com/kikuchipy/kikuchipy>`_ in order to make changes
+to KikuchiPy.
+
+Make a local copy of your forked repository and change directories::
+
+    git clone https://github.com/your-username/kikuchipy.git
+    cd kikuchipy
+
+Set the ``upstream`` remote to the main KikuchiPy repository::
+
+    git remote add upstream https://github.com/kikuchipy/kikuchipy.git
+
+Install the required dependencies while making the development version available
+globally::
+
+    pip install -e .[dev]
+
+This installs all necessary development dependencies, including those for
+running tests and building documentation.
+
+The code making up KikuchiPy is formatted closely following the `Style Guide for
+Python Code <https://www.python.org/dev/peps/pep-0008/>`_ with `The Black Code
+style <https://black.readthedocs.io/en/stable/the_black_code_style.html>`_. We
+use `pre-commit <https://pre-commit.com>`_ to run ``black`` automatically prior
+to each local commit. Please install it in your environment::
+
+    pre-commit install
+
+Next time you commit some code, your code will be formatted inplace according
+to our `black configuration
+<https://github.com/kikuchipy/kikuchipy/blob/master/pyproject.toml>`_.
+
+Note that ``black`` won't format `docstrings
+<https://www.python.org/dev/peps/pep-0257/>`_. We follow the `numpydoc
+<https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard>`_
+standard.
+
+.. _making-changes:
+
+Making changes
+==============
+
+Create a new feature branch::
+
+    git checkout master -b your-awesome-feature-name
+
+When you've made some changes, with them with::
+
+    git status
+
+Add and commit your created, modified or deleted files::
+
+   git add my-file-or-directory
+   git commit -s -m "An explanatory commit message"
+
+The ``-s`` makes sure that you sign your commit with your `GitHub-registered
+email <https://github.com/settings/emails>`_ as the author. You can set this up
+following `this GitHub guide
+<https://help.github.com/en/github/setting-up-and-managing-your-github-user-account/setting-your-commit-email-address>`_.
+
+.. _keeping-your-branch-up-to-date:
+
+Keeping your branch up-to-date
+==============================
+
+Switch to the ``master`` branch::
+
+   git checkout master
+
+Fetch changes and update ``master``::
+
+   git pull upstream master --tags
```

Comment:
Yes, it is a quite nice feature. E.g., pyXem has made available snapshots of the `master` branch for different pre-release versions, so by doing:

```bash
git pull upstream master --tags
git checkout tags/v0.5
```

I can check out the state of pyXem all the way back to the 0.5 release, instead of only relying on the `master` branch. This is nice for at least historical reasons.